### PR TITLE
Integrate Allure HTML reporting for testing

### PR DESCRIPTION
**Pull request by jiridanek**
Added Allure reporting dependencies and configuration for enhanced test result reporting. Updated code annotations for issue tracking, silenced internal Allure error logging during tests, and documented Allure report generation in README.

image

### CODE REVIEW

File: src/test/resources/allure.properties [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#
+# Copyright Skodjob authors.
+# License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+#
+
+allure.results.directory=target/allure-results
+
+allure.link.issue.pattern = https://issues.redhat.com/browse/{}
+allure.link.tms.pattern = https://polarion.engineering.redhat.com/polarion/#/project/OpenDataHub/workitem?id={}
```
Comment (kornys) [inline src/test/resources/allure.properties:9]: Are you kidding?

File: src/test/resources/allure.properties [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#
+# Copyright Skodjob authors.
+# License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+#
+
+allure.results.directory=target/allure-results
+
+allure.link.issue.pattern = https://issues.redhat.com/browse/{}
```
Comment (kornys) [inline src/test/resources/allure.properties:8]: Why do you put redhat url into upstream project?

File: src/test/resources/allure.properties (hunk 1 above)
Comment (jiridanek) [reply src/test/resources/allure.properties]: removed

File: src/test/resources/allure.properties (hunk 2 above)
Comment (jiridanek) [reply src/test/resources/allure.properties:8]: this tracker is used in upstream project, https://github.com/search?q=org%3Aopendatahub-io%20%22issues.redhat.com%22&type=code

File: README.md [hunk 3]
```diff
@@ -86,3 +86,7 @@ The docs are generated with every build and the changes should be committed when
 
 The plugin is still under development so the format could change.
 For more info see the plugin repository on GitHub.
+
+## Allure reports
```
Comment (kornys) [inline README.md:90]: Use block of code instead of just highlight

**Review by kornys — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (jiridanek) [reply README.md:90]: done

**Review by kornys — APPROVED** [review]

**Review by jstourac — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

**Pull request merged**